README.md

```diff
@@ -1,52 +1,76 @@
-
-## Documentation
-
-Project documentation, circuit definitions, API references etc can be found at https://verify.mina.nil.foundation/docs.
-
-## Auxiliary Proof Generator (`aux-proof-gen`)
-
-Auxiliary proof generator is UNIX-style application taking Mina Protocol state as an input and producing auxiliary proof as an output. 
-
-The generator prototype is implemented in C++ and uses =nil; Crypto3 C++ Cryptography Suite (https://github.com/nilfoundation/crypto3) for cryptographic primitives definition.
-
-### Dependencies
-
-Libraries requirements are as follows:
-* Boost (https://boost.org) (>= 1.76)
-
-Compiler/environment requirements are as follows:
-* CMake (https://cmake.org) (>= 3.13)
-* GCC (>= 10.3) / Clang (>= 9.0.0) / AppleClang (>= 11.0.0)
-
-### Building
-
-#### Native
-
-`mkdir build && cd build && cmake .. && make aux-proof-gen`
-
-### Usage
-
-`bin/aux-proof-gen/aux-proof-gen ../share/mina/genesis_proof_*`
-
-### Tests
-
-`make zk_lpc_test && make zk_fri_test`
-
-### Benchmarks
-
-`make zk_lpc_performance_test`
-
-## Community
-
-Issue reports are preferred to be done with Github Issues in here: https://github.com/nilfoundation/evm-mina-verification/issues.
-
-Forum-alike discussion topics are better to be done with Discussions section in here: https://github.com/NilFoundation/evm-mina-verification/discussions
-
-Usage and development questions a preferred to be asked in a Telegram chat: https://t.me/nilfoundation
+# Mina State proof verification in EVM
+
+This repository presents the solidity smart contracts required to verify Mina ledger proof in EVM. 
+The repository uses Hardhat as development environment for compilation, testing and deployment tasks.
+
+## Dependencies
+
+- [Hardhat](https://hardhat.org/)
+- [nodejs](https://nodejs.org/en/) >= 16.0
+
+## Compile 
+```
+npx hardhat compile
+```
+
+## Test
+```
+npx hardhat test #Execute tests
+REPORT_GAS=true npx hardhat test # Test with gas reporting
+```
+
+## Deploy
+
+Launch ganche using the following
+```
+ganache-cli -l 900000000 -m 'test test test test test test test test test test test junk' -g 20000 --verbose
+```
+ 
+To deploy to test environment (ex: Ganache)
+```
+npx hardhat deploy  --network ganache 
+```
+
+Hardhat re-uses old deployments, to force re-deploy add the `--reset` flag above
+
+## Usage
+
+Below two tasks execute flows to validate ledger state and validate account state. 
+Please note , these work against the above delpoyment, hence , you must run the deploy before
+executing the following.
+
+### Validate Ledger State
+```
+npx hardhat validate_ledger_state --proof ./test/data/proof_v.data --ledger jwYPLbRQa4X86tSJs1aTzusf3TNdVTj58oyWJQB132sEGUtKHcB  --network ganache
+```
+Inputs
+- _proof : This is file path with the full mina ledger state 
+proof retrieved from proof market._
+- _ledger: This is the hash of the ledger which this proof attests._
+- _network: Network to run this task against_
```

Comment:
"This is" part is not required when you describe inputs.
In markdown lists, it's better to indent wrapped lines to a common level.
Note the punctuation in list items: it should be consistent.
Mina is a proper noun, so mind the capitalization.
Composite constructions with a specific meaning, like "ledger state proof",
are better left on a single line and not wrapped.
It's easier to read them this way.
Proof market is the particular proof market that we're developing,
so it's *the* proof market.


```suggestion
- _proof: file path with the full Mina ledger state proof
 retrieved from the proof market._
- _ledger: hash of the ledger to which this proof attests._
- _network: network to run this task against._
```